app/models/adviser.rb

```diff
@@ -48,12 +57,21 @@ def field_order
 
   private
 
+  def check_for_changes
+    @old_firm_id = firm_id_change.first if firm_id_changed?
+  end
```

Comment:
Yes as discussed, all record of what changed is gone by the time we get to the after_commit hook.